src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceCounterCategory.cs

```diff
@@ -475,7 +475,7 @@ internal static string[] GetCounterInstances(string categoryName, string machine
 
             string[] instanceNames = new string[categorySample._instanceNameTable.Count];
             categorySample._instanceNameTable.Keys.CopyTo(instanceNames, 0);
-            if (instanceNames.Length == 1 && instanceNames[0].CompareTo(PerformanceCounterLib.SingleInstanceName) == 0)
+            if (instanceNames.Length == 1 && instanceNames[0] == PerformanceCounterLib.SingleInstanceName)
```

Comment:
Note: `string.CompareTo` does a culture-sensitive comparison based on the current culture. I don't think that was the intention here checking equality with this constant, so I've fixed it to do an ordinal comparison.